boot/freeldr/freeldr/lib/peloader.c

```diff
@@ -624,35 +624,41 @@ PeLdrAllocateDataTableEntry(
         FrLdrHeapFree(DataTableEntry, TAG_WLDR_DTE);
         return FALSE;
     }
-    RtlZeroMemory(Buffer, Length);
 
-    DataTableEntry->BaseDllName.Length = Length;
-    DataTableEntry->BaseDllName.MaximumLength = Length;
-    DataTableEntry->BaseDllName.Buffer = PaToVa(Buffer);
+    // Save Buffer, in case of later failure.
+    BaseDllNameBuffer = Buffer;
+
+    RtlZeroMemory(Buffer, Length);
```

Comment:
Shall I remove zeroing, which seems useless to me?

Reply:
Just keep all that as it is.